src/seedu/addressbook/commands/Command.java

```diff
@@ -15,6 +15,14 @@
     protected AddressBook addressBook;
     protected List<? extends ReadOnlyPerson> relevantPersons;
     private int targetIndex = -1;
+    protected boolean mutated = false;
+
+    /**
+     *  @return true for command types that mutate the data
+     */
+    public boolean isMutating() {
+        return mutated;
```

Comment:
Having a default implementation for the `isMutating` method helps to reduce the amount of code. However, some may argue that choosing `false` (or `true`) as the default is arbitrary and hurts the intuitiveness of the code. Making the method `abstract` forces all child classes to declare whether it is a mutating command or not, rather than depend on an arbitrary decision made in the parent class.